http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/JsonPayloadIT.java

```diff
@@ -0,0 +1,189 @@
+                .then()
+                .statusCode(CREATION_STATUS_CODE)
+                .body(containsString("619 Rockaway Parkway, Hillsboro, California, 7507"));
+
+        getApp().given()
+                .get("/persons")
+                .then()
+                .statusCode(OK_STATUS_CODE)
+                .body(containsString("Hello, Singleton Middleton! You have 4 unread messages."));
+
+    }
+
+    @Test
+    public void sendJSONWithEmptyFieldsPayload() throws IOException {
+        String jsonPayload = readJSONFile("football-empty-teams.json");
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(BAD_REQUEST_STATUS_CODE)
+                .body(containsString("Some fields are empty or null"));
+    }
+
+    @Test
+    public void sendInvalidJSONPayload() throws IOException {
+        String jsonPayload = readJSONFile("football-teams_invalid.json");
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(BAD_REQUEST_STATUS_CODE)
+                .body(containsString(
+                        "Invalid JSON - JsonProcessingException : Unexpected character ('{' (code 123)): was expecting comma to separate Array entries"));
+    }
+
+    @Test
+    public void sendASingleJSONObject() throws JsonProcessingException {
+        FootballTeam teamX = new FootballTeam(
+                "Single Team",
+                "Pink",
+                3,
+                new Date(1900, 1, 1),
+                "A",
+                Arrays.asList("You", "You"));
+        ObjectMapper objectMapper = new ObjectMapper();
+        String jsonPayload = objectMapper.writeValueAsString(teamX);
+
+        getApp().given()
+                .contentType(ContentType.JSON)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(BAD_REQUEST_STATUS_CODE)
+                .body(containsString(
+                        "JsonMappingException : Cannot deserialize value of type `java.util.HashSet<io.quarkus.ts.http.advanced.reactive.FootballTeam>`"));
+
+    }
+
+    @Test
+    public void sendIncorrectContentType() throws IOException {
+        String jsonPayload = readJSONFile("football-empty-teams.json");
+
+        getApp().given()
+                .contentType(ContentType.TEXT)
+                .body(jsonPayload)
+                .when()
+                .post("/football/upload-football-json")
+                .then()
+                .statusCode(UNSUPPORTED_MEDIA_TYPE_STATUS_CODE);
+
+    }
+
+    @Test
+    public void sendSpecialCharactersJSONPayload() throws Exception {
```

Comment:
This test is sending plain text (it is a `String`) via client, received `String` in REST endpoint. Special characters or not, you are testing your own instance of `ObjectMapper`.

As far as extensions deserializers are concerned - none used
As far as extension serializers are concerned - `teams` should be serialized, but that is not tested at all, what is tested is that text containes `Teams added successfully`. So you are not testing whether special characters are handled properly by serializers, but you are testing that no exception is raised. If there was any encoding issue, you wouldn't know.